Testscripts/Windows/VM-HOT-RESIZE.ps1

```diff
@@ -20,8 +20,28 @@ function Main {
 
         $loadBalanceName = (Get-AzureRmLoadBalancer -ResourceGroupName $AllVMData.ResourceGroupName).Name
         $VirtualMachine = Get-AzureRmVM -ResourceGroupName $AllVMData.ResourceGroupName -Name $AllVMData.RoleName
-        for ( $i = 0; $i -le 30; $i++ ) {
+        foreach ($param in $currentTestData.TestParameters.param) {
+            if ($param -match "TestMode") {
+                $testMode = $param.Replace("TestMode=","").Replace('"',"")
+            }
+        }
+        if ($testMode -eq "economy") {
+            Write-LogInfo "The test mode is economy mode"
+            $toalTestTimes = 10
+        } else {
+            $toalTestTimes = 30
+        }
+
+        for ( $i = 0; $i -le $toalTestTimes; $i++ ) {
             $vmSizes = (Get-AzureRmVMSize -ResourceGroupName $AllVMData.ResourceGroupName -VMName $AllVMData.RoleName).Name
+            # For economy mode, select a size by random in order to cover as many different serial sizes as possible
+            if ($testMode -eq "economy") {
+                $vmSizes = $vmSizes | Get-Random -Count 1
```

Comment:
this is not the design discussed in the PR.
The suggestion there was to test the smallest and the largest VM size from each SKU. So testing at least the bigger sizes would be helpful to catch any large specs issues.

here are you taking a random size from each VM size SKU?

Reply:
Yes, I take a random size to test. 
In my opinion, it is difficult to test the smallest and largest VM size  from each SKU because the resize is **dynamic** change according to the VM size.  Assume that a VM initial size is X, and it supports to resize to Y_min and Y_max. However, the Y_min/Y_max support to resize to Z1 and Z2.